felt/src/lib.rs

```diff
@@ -40,16 +38,641 @@ pub trait FeltOps {
+        self.value *= &rhs.value;
+    }
+}
+
+impl Pow<u32> for Felt {
+    type Output = Self;
+    fn pow(self, rhs: u32) -> Self {
+        Self {
+            value: self.value.pow(rhs),
+        }
+    }
+}
+
+impl<'a> Pow<u32> for &'a Felt {
+    type Output = Felt;
+    fn pow(self, rhs: u32) -> Self::Output {
+        Self::Output {
+            value: (&self.value).pow(rhs),
+        }
+    }
+}
+
+impl Div for Felt {
+    type Output = Self;
+    fn div(self, rhs: Self) -> Self {
+        Self {
+            value: self.value / rhs.value,
+        }
+    }
+}
+
+impl<'a> Div for &'a Felt {
+    type Output = Felt;
+    fn div(self, rhs: Self) -> Self::Output {
+        Self::Output {
+            value: &self.value / &rhs.value,
+        }
+    }
+}
+
+impl<'a> Div<Felt> for &'a Felt {
+    type Output = Felt;
+    fn div(self, rhs: Self::Output) -> Self::Output {
+        Self::Output {
+            value: &self.value / rhs.value,
+        }
+    }
+}
+
+impl Rem for Felt {
+    type Output = Self;
+    fn rem(self, rhs: Self) -> Self {
+        Self {
+            value: self.value % rhs.value,
+        }
+    }
+}
+
+impl<'a> Rem<&'a Felt> for Felt {
+    type Output = Self;
+    fn rem(self, rhs: &Self) -> Self {
+        Self {
+            value: self.value % &rhs.value,
+        }
+    }
+}
+
+impl Zero for Felt {
+    fn zero() -> Self {
+        Self {
+            value: FeltBigInt::zero(),
+        }
+    }
+
+    fn is_zero(&self) -> bool {
+        self.value.is_zero()
+    }
+}
+
+impl One for Felt {
+    fn one() -> Self {
+        Self {
+            value: FeltBigInt::one(),
+        }
+    }
+
+    fn is_one(&self) -> bool {
+        self.value.is_one()
+    }
+}
+
+impl Bounded for Felt {
+    fn min_value() -> Self {
+        Self {
+            value: FeltBigInt::min_value(),
+        }
+    }
+
+    fn max_value() -> Self {
+        Self {
+            value: FeltBigInt::max_value(),
+        }
+    }
+}
+
+impl Num for Felt {
+    type FromStrRadixErr = ParseFeltError;
+    fn from_str_radix(string: &str, radix: u32) -> Result<Self, Self::FromStrRadixErr> {
+        Ok(Self {
+            value: FeltBigInt::from_str_radix(string, radix)?,
+        })
+    }
+}
+
+impl Integer for Felt {
+    fn div_floor(&self, rhs: &Self) -> Self {
+        Self {
+            value: self.value.div_floor(&rhs.value),
+        }
+    }
+
+    fn div_rem(&self, other: &Self) -> (Self, Self) {
+        let (div, rem) = self.value.div_rem(&other.value);
+        (Self { value: div }, Self { value: rem })
+    }
+
+    fn divides(&self, other: &Self) -> bool {
+        self.value.divides(&other.value)
+    }
+
+    fn gcd(&self, other: &Self) -> Self {
+        Self {
+            value: self.value.gcd(&other.value),
+        }
+    }
+
+    fn is_even(&self) -> bool {
+        self.value.is_even()
+    }
+
+    fn is_multiple_of(&self, other: &Self) -> bool {
+        self.value.is_multiple_of(&other.value)
```

Comment:
It's a little more complicated than this in fields.
```
P = 5
2.is_multiple_of(3) == true
```
This is because `3 * 4 == 12` is `2` in the field `5`.